pkg/generic/generic.go

```diff
@@ -32,13 +31,17 @@ import (
 type Generic interface {
 	Closer
 	// PayloadCodec return codec implement
+	// this is used for generic which does not need IDL
 	PayloadCodec() remote.PayloadCodec
 	// PayloadCodecType return the type of codec
 	PayloadCodecType() serviceinfo.PayloadCodec
 	// RawThriftBinaryGeneric must be framed
 	Framed() bool
-	// GetMethod to get method name if need
+	// GetMethod is to get method name if needed
 	GetMethod(req interface{}, method string) (*Method, error)
+	// CodecInfo return codec info
+	// this is used for generic which needs IDL
+	CodecInfo() serviceinfo.CodecInfo
```

Comment:
Is it ok to add a new method to Generic interface?

Reply:
Maybe add to a new interface Codecor interface{ CodecInfo() serviceinfo.CodecInfo }, and type assert is in generic Call

Reply:
all the generic types are hidden in different package from generic call's, so it seems to be hard to adding a separate interface.